### DISCUSSION TITLE
fr/modes/val/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#18296] [Docs]
# fr/modes/val/

Apprenez à valider votre modèle YOLO11 à l'aide de mesures précises, d'outils faciles à utiliser et de paramètres personnalisés pour des performances optimales.

https://docs.ultralytics.com/fr/modes/val/

**Comment by slhermite** [comment]
hey ! 

mAP50 is defined as the area under the PR curve, calculated for different confidence thresholds, with an IoU threshold fixed at 0.5.

In theory, when running *model.val(..., conf=conf)*, the mAP50 result should remain unchanged regardless of the conf parameter value. However, that’s not what I’m observing...

I don’t understand why this happens. Could you clarify?

**Reply by glenn-jocher** [reply]
@slhermite thank you for your question! The `conf` parameter affects which detections are included in the evaluation. Lowering it allows more detections, including lower-confidence ones, to pass the threshold, which can influence metrics like mAP50 since the precision-recall curve is affected by the number of detections. This is expected behavior as the PR curve and consequently mAP are recalculated based on the filtered predictions. For more insights, refer to the [Validation Guide](https://docs.ultralytics.com/modes/val/).

**Reply by slhermite** [reply]
Thank you for responding so quickly. I have reviewed the documentation, and this is exactly why I find your response inconsistent.

The PR curve (Precision-Recall) is defined by calculating precision and recall for all possible confidence thresholds, ranging from 0 to 1. At each threshold, we filter a more or less significant number of detections, which causes precision and recall to vary. This allows us to plot the curve. Then, we calculate the area under this curve, which gives us the AP (Average Precision), or mAP50 if we set the IoU threshold to 50.

Your response suggests that there would be a preliminary step where detections are filtered, which seems to contradict the very definition of the PR curve. Since it is by filtering detections with the confidence threshold that we are able to construct the curve...

**Reply by slhermite** [reply]
Ok, I looked at your scripts and you filter a part of the detections before calculating the precision-recall curve...

Thank you to the Ultralytics team for redefining mathematical concepts with your unique vision of the world. The scientific community is truly grateful. Long live diversity of thought!

**Reply by glenn-jocher** [reply]
The `conf` parameter adjusts the minimum confidence threshold for detections, which directly influences the precision-recall calculations by filtering detections before they are included in the PR curve computation. This is a design choice to enhance practical usability and efficiency. For a more detailed breakdown, please refer to the [Validation Guide](https://docs.ultralytics.com/modes/val/). If you have further concerns, feel free to share!

**Comment by UltralyticsAssistant** [comment]
👋 Bonjour, merci pour votre intérêt envers Ultralytics 🚀! Nous vous recommandons de consulter la [documentation](https://docs.ultralytics.com/fr) pour obtenir des informations détaillées sur l'utilisation, notamment pour [valider votre modèle YOLO](https://docs.ultralytics.com/fr/modes/val/). Ces ressources pourraient répondre à la plupart des questions fréquentes.

Si ceci est un 🐛 **rapport de bogue**, veuillez fournir un [exemple minimal reproductible](https://docs.ultralytics.com/help/minimum_reproducible_example/) afin que nous puissions enquêter dessus plus efficacement.

Si vous avez une ❓ **question sur un entraînement personnalisé**, n'hésitez pas à partager le maximum d'informations, comme des exemples d'images de jeu de données, des journaux d'entraînement, et à vérifier que vous suivez nos [conseils pour de meilleurs résultats d'entraînement](https://docs.ultralytics.com/guides/model-training-tips/).

Rejoignez la communauté Ultralytics là où cela vous convient le mieux ! Pour des conversations en temps réel, passez sur notre [Discord](https://discord.com/invite/ultralytics) 🎧. Pour des discussions plus approfondies, explorez notre [Discourse](https://community.ultralytics.com). Vous pouvez également partager vos connaissances sur le [Subreddit](https://reddit.com/r/Ultralytics).

## Mise à jour

Mettez à jour le package `ultralytics`, ainsi que toutes les [dépendances requises](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), dans un environnement [**Python>=3.8**](https://www.python.org/) avec [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) afin de vérifier que votre problème n’est pas déjà résolu dans la dernière version :

```bash
pip install -U ultralytics
```

## Environnements pris en charge

YOLO peut être exécuté dans les environnements vérifiés suivants (avec toutes les dépendances, y compris [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) et [PyTorch](https://pytorch.org/) préinstallés) :

- **Notebooks** avec GPU gratuit : <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Consultez le [guide de démarrage rapide GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Consultez le [guide de démarrage rapide AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Image Docker**. Consultez le [guide de démarrage rapide Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Statut

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Si ce badge est vert, tous les tests [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) passent actuellement. Ces tests vérifient le bon fonctionnement de tous les [Modes](https://docs.ultralytics.com/modes/) et [Tâches](https://docs.ultralytics.com/tasks/) YOLO sur macOS, Windows et Ubuntu toutes les 24 heures et à chaque commit.

---

🤖 Ceci est une réponse automatique, mais ne vous inquiétez pas, un ingénieur d'Ultralytics examinera votre message et vous assistera bientôt ! 😊
